### DISCUSSION TITLE
It is downloading the .dev version instead of 2.7.9[Question]

### DISCUSSION THREAD

**Discussion by leandro-pilz** [hm21/pro_image_editor#92] [Q&A] [answered]
It is downloading the .dev version instead of 2.7.9 when informing in flutter, that is, I inform 2.7.9 and download 2.7.10-dev.1

**Comment by hm21** [comment]
Interesting, so in your pubspec.yaml file you have `pro_image_editor: 2.7.9`, but it installs version `2.7.10-dev.1`, right? 
Can you try to run `flutter clean` first and check if it installs the correct version.
And how do you know that the dev version is installed and not 2.7.9?

**Comment by leandro-pilz** [comment]
Right yes, I've already run flutter clean and then flutter pub get, as I know you're downloading the version I mentioned, follow below

![Captura de Tela 2024-05-17 às 08 51 34](https://github.com/hm21/pro_image_editor/assets/60925091/39b4ce1e-9d17-498c-aa44-ef6c363acae7)

![Captura de Tela 2024-05-17 às 08 52 59](https://github.com/hm21/pro_image_editor/assets/60925091/0008bfe1-66e5-455f-93e8-6ab6f0e71b99)

**Comment by hm21** [comment]
Okay, I got it. I also tried to reproduce it, but in my case it always takes the correct version. It's also special that it downloads the version `2.7.10-dev.1` because this is an older version from the dev branch. I already released in the dev branch version `3.0.0-dev.3`. Anyway, are you using flutter `3.22.0` or an older version of it? And do you use the `stable` channel from flutter?

**Comment by leandro-pilz** [comment]
I'm currently using version 3.19.6 of flutter, the question is I have a lib of mine which is a ui package so that my app uses everything from this lib, so I'm setting the pro editor in my ui package and pp consuming from there, and that's why the problem seems to be occurring. I'll try to use it directly in the app even without going through the UI package.

**Comment by hm21** [comment]
Sounds interesting. Is your UI package public or private? If it's public, I can test it if I have the same problem. 
Just to confirm, you're also running `flutter clean` inside your UI package and not using the `2.7.10-dev.1` version there, right?

**Comment by leandro-pilz** [comment]
Exactly, I'm using 2.7.8 in my UI package, I apply flutter clean and flutter pub get, the package is private, as it is a company application, we work with monitoring works and we need a photo editor, I found the its very good indeed.

**Comment by leandro-pilz** [comment]
I did all the flutter clean and flutter pub get steps and when running the project it ends up arriving in the buld and fails
![Captura de Tela 2024-05-17 às 15 07 43](https://github.com/hm21/pro_image_editor/assets/60925091/e9ffcfe6-fa66-4d42-a9a7-3455b84d5e11)

**Comment by hm21** [comment]
So you now install version 2.7.8 in your project, and inside your UI package you still have the editor installed? Because it seems like there are different versions of the editor in conflict, or maybe there is a conflict with the `image` package. Are you also installing the `image` package inside your current package or UI package? If so, which version are you using? And when you run flutter pub get, do you see any warnings/errors in the terminal.

**Comment by leandro-pilz** [comment]
Hi, perfect, in fact it was the image version, yesterday at the end of the day I ended up detecting this, I really appreciate your help and your commitment to helping me. The package is very good, congratulations on the excellent work.

I'm just having difficulty using it within my UI package, I'm going to use it in the application right now, what happens is, when I close the editor, the loading of saving changes remains on the screen, if I put a Navigator.of(context ).pop() in the onCloseEditor method it ends up closing the current screen, and the loading remains, I tried putting Navigator.of(context).pop() in onImageEditingComplete and the same thing happens, remembering that this behavior occurs when using the pro editor In my UI package, the app works normally.

**Comment by leandro-pilz** [comment]
Bom amigo desculpa estar tirando seu tempo, mas agora a unica coisa que ocorre, é, tenho o código certinho, porém ao clicar em salvar a alteração da image, o pro editor fecha, ou seja retorna, mas o doneLoading fica aberto na tela da minha aplicação, saberia me dizer porque?  Segue o código chamador do pro editor(componente)
![image](https://github.com/hm21/pro_image_editor/assets/60925091/abe709cd-9b69-46d9-943d-8d1374fadc1c)

Fiz um app teste e nele funciona perfeitamente.

**Comment by hm21** [comment]
In most cases, the problem is that people are using a special navigation system like the one in the `get` package. 
The other case is that something is wrong with the configurations. Can you post me the ProImageEditor widget, so I can take a look at it, maybe I can see the issue then. For example, `ProImageEditor.network(/*config stuff*/)`.

**Comment by leandro-pilz** [comment]
Ola amigo, realmente implementei o pro image editor em um outro projeto para teste utilizando a minha biblioteca de ui contendo o seu projeto e tudo funcionou normalmente.  No app da empresa em principio usamos no Navigator, mas vou verificar todo o fluxo para ver onde está o problema.  Agradeço a sua ajuda.  

Uma dica apenas, quero mudar a cor do systemNavigationBarColor mas em iOS e não consigo, não surte o efeito desejado. segue abaixo o código utilizado para fazer isto:

![Captura de Tela 2024-05-20 às 08 54 47](https://github.com/hm21/pro_image_editor/assets/60925091/cbd88395-b140-4b5a-b747-31c96caf1ae8)

**Comment by hm21** [comment]
Ola amigo, 

Inside the code I use `AnnotatedRegion<SystemUiOverlayStyle>` to set these settings. You can try to edit your configs from the image editor like below that should work:

```dart
ProImageEditor.asset(
    'demo.png',
    configs: ProImageEditorConfigs(
      imageEditorTheme: ImageEditorTheme(
        uiOverlayStyle: SystemUiOverlayStyle.dark.copyWith(
            // your configs...
            ),
      ),
  ),
```

**Comment by leandro-pilz** [comment]
Perfect, good my friend, as I reported above, using the pro editor through the UI library in a test project, it worked, but in the app project the reported problem still persists, I know it is already helping me a lot, but there would be a way to have more direct contact, to talk and be able to show the code and maybe then you can help me maybe detect the problem, which in fact I can't understand what happens and only in the app itself.

I checked the route issue and we only use Flutter's navigator, it doesn't have any pakage like get for navigation.

Thank you in advance for all your help.

**Comment by leandro-pilz** [comment]
I identified the problem, we used the package
https://pub.dev/packages/flutter_screenutil
to be able to adapt the screen to tablet issues and others, and otherwise it works, but then we have the problem of responsiveness for tablets.

**Comment by hm21** [comment]
Okay, this is interesting. I will try it in my example when I have time and if I can reproduce it, I will also try to fix it. Do I need to configure the `ScreenUtilInit` in a special way, or is it enough to just wrap all content in it?

**Comment by hm21** [comment]
I just tried to reproduce it, but in my case it still works even with this package. Can you post here for me a simple example code how I can reproduce it.

**Comment by leandro-pilz** [comment]
Hello, friend, first of all I apologize, because in reality it is not this package that causes the problems, but this one, asuka

https://pub.dev/packages/asuka

I will send the code we have with both packages in the main configuration
![image](https://github.com/hm21/pro_image_editor/assets/60925091/b4d8b353-45c0-4478-9c29-b2b24fd9b4fe)

When removing the line "child: Asuka.builder(context, widget)", from the image below, just returning the builder "widget", it works normally.

I'm on flutter version 3.19.6 with editor version 2.7.8

**Comment by hm21** [comment]
Okay, I think I found the problem. The package `Asuka` needs the build context from the dialog itself to close the same dialog. I updated the loading-dialog that it first tries to use the own context before it takes the global context to close it. 
I released version `2.7.10`, can you try it if it works with this version?

**Comment by leandro-pilz** [comment]
Hello, of course I will do the test, but I use flutter 3.19.6, no problems?

Em ter., 21 de mai. de 2024 às 10:52, Alex Frei ***@***.***>
escreveu:

**Comment by hm21** [comment]
Oops I forgot that you said you use 3.19.6🙈 Actually I only support the newest flutter version, but I released for you version `2.7.8-fix-1` now which also supports 3.19.6.

**Comment by leandro-pilz** [comment]
Thank you very much, from what I saw it worked well, I will do more tests.

**Comment by leandro-pilz** [comment]
The adjustments worked, the only issue is that the top and bottom bars are
taking on the color of my app, before they were black. Sorry to disturb
you, but this point remains, which I believe to get around it.

[image: WhatsApp Image 2024-05-21 at 13.23.13.jpeg]

Em ter., 21 de mai. de 2024 às 11:22, Alex Frei ***@***.***>
escreveu:

**Comment by hm21** [comment, marked as answer]
Can you check inside the `ProImageEditorConfigs` and then `imageEditorTheme`  if there is set any colors for the appbar? The code in the editor looks like that:

```dart
AppBar(
      automaticallyImplyLeading: false,
      foregroundColor: imageEditorTheme.appBarForegroundColor,
      backgroundColor: imageEditorTheme.appBarBackgroundColor,
      /// other stuff
)
```

**Comment by leandro-pilz** [comment]
Very good friend, perfect, resolved, I really appreciate all the attention
and help. Very good to the editor

Em ter., 21 de mai. de 2024 às 13:38, Alex Frei ***@***.***>
escreveu:
